src/dune_rules/install_rules.ml

```diff
@@ -174,12 +177,17 @@ end = struct
           List.map ~f:(fun (cm_kind, p) -> (cm_dir m cm_kind, p))
         in
         let modules_impl =
+          let make_cmt ~cm_kind m ml_kind =
+            let open Option.O in
+            let+ cmt =
+              Obj_dir.Module.cmt_file obj_dir m ~ml_kind
+                ~cm_kind:(Ocaml cm_kind)
+            in
+            (cm_kind, cmt)
+          in
           List.concat_map installable_modules.impl ~f:(fun m ->
               common m
-              @ List.filter_map Ml_kind.all ~f:(fun ml_kind ->
-                    let open Option.O in
-                    let+ cmt = Obj_dir.Module.cmt_file obj_dir m ~ml_kind in
-                    (Cm_kind.Cmi, cmt))
+              @ List.filter_map Ml_kind.all ~f:(make_cmt ~cm_kind:Cmi m)
```

Comment:
do we need these changes? They seem like unrelated refactoring to me.

Reply:
good point, simplified in https://github.com/jchavarri/dune/pull/4/commits/ff572a3ea22c63104d5ff9952fd82c7a9a320b20.